### PR TITLE
simplecv: GPU MANO + logging perf for exoego visualize

### PR DESCRIPTION
**Pull request by pablo-rerun**
## What

Performance wins for `visualize_exo_ego` (the exo/ego Rerun logger). No behavior change beyond float-epsilon.

- **GPU/torch MANO.** The per-frame MANO LBS forward now runs through `MANOLayerTorch` on the available device (CUDA when present, otherwise CPU) instead of the numpy layer — one runtime path instead of numpy+torch. Measured numpy vs torch: **torch-CPU ~2.8×**, **torch-CUDA ~68×** faster, parity ~1e-7. `MANOLayerTorch` gains a `use_pca` arg (mirrors `MANOLayerNP`) and a numpy `faces` accessor so it fully replaces the numpy layer here.
- **Vectorized nanmean.** `_ConfidenceAwareColumnList.partition` computed a per-frame average-confidence scalar with a Python loop calling `np.nanmean` per frame; replaced with a vectorized `_segment_nanmean` (uniform-length reshape fast path + `reduceat` fallback). **~8× faster** at the hot 133-keypoints-per-frame size, parity ~1e-7.
- **gRPC buffer knob.** `RerunTyroConfig.server_memory_limit` (default `"4GB"`) so long multi-camera recordings (e.g. a 46-min EPFL session) don't silently drop history from the viewer's ~1 GiB default buffer.

## Numbers

End-to-end `visualize_exo_ego` on EPFL `YH2040/2023_09_12_09_50_41` (~23k frames, CUDA, warm): **~16s → ~10s**.

## Verification

`pixi run -e simplecv-dev --frozen {lint,typecheck,deadcode,tests}` — ruff clean, pyrefly 0 errors, vulture clean, full suite **139 passed / 27 skipped**. `_segment_nanmean` equivalence and numpy↔torch MANO parity checked with hypothesis + a parity benchmark.

## Not in this PR

The EPFL ego "black video + camera desync" **bug fix** (HoloLens dying mid-session) is intentionally split into a separate follow-up PR.

### CODE REVIEW

File: packages/simplecv/simplecv/apis/view_exoego.py [hunk 1]
```diff
@@ -348,25 +352,30 @@ def log_mano_batch(
...
-            hand_idx: ndarray = RIGHT_HAND_IDX if mano_layer.side == "right" else LEFT_HAND_IDX
-            xyz_coco_mano[:, hand_idx, :] = xyz_mano_np[0:n_frames_mano_total]
-            conf_coco_mano[:, hand_idx] = 1.0
+            with torch.no_grad():
+                verts_torch: Float32[torch.Tensor, "n_frames n_verts=778 3"]
+                joints_torch: Float32[torch.Tensor, "n_frames n_joints=21 3"]
+                verts_torch, joints_torch = mano_layer(
+                    torch.from_numpy(np.ascontiguousarray(poses)).float().to(mano_device),
+                    torch.from_numpy(np.ascontiguousarray(translations)).float().to(mano_device),
+                )
```
[the remark this answers is not in the document]
Comment (pablovela5620) [reply packages/simplecv/simplecv/apis/view_exoego.py:364]: Good catch — fixed in 1662c2b: `poses`/`translations` are now sliced to `n_frames_mano_total` before the forward. Behavior-preserving (the output slices downstream become no-ops), so it's pure savings — less CPU→GPU transfer/compute and lower peak GPU memory — when the MANO stream is longer than the label timeline.
